### PR TITLE
Rake Task to Fill in license_normalized Values for Projects

### PR DESCRIPTION
**Pull request by mikeyoung85**
I still need to figure out the best way to run this task.

### CODE REVIEW

File: lib/tasks/projects.rake [hunk 1]
```diff
@@ -135,4 +135,19 @@ namespace :projects do
     number_to_sync = args.number_to_sync || 2000
     Project.least_recently_updated_stats.where(platform: supported_platforms).limit(number_to_sync).each{|project| project.update_maintenance_stats_async(priority: :low)}
   end
+
+  desc 'Set license_normalized flag'
+  task set_license_normalized: :environment do
+    supported_platforms = ["Maven", "NPM", "Pypi", "Rubygems", "Nuget", "Packagist"]
```
Comment (havocp) [inline lib/tasks/projects.rake]: assuming these are the right capitalizations that libraries uses

File: lib/tasks/projects.rake [hunk 2]
```diff
@@ -135,4 +135,19 @@ namespace :projects do
...
+    Project.where(platform: supported_platforms, license_normalized: false).find_in_batches do |group|
+      group.each do |project|
+        project.normalize_licenses
+        # check if we set a new value
+        if project.license_normalized_changed?
+          # update directly to skip any callbacks
```
Comment (havocp) [inline lib/tasks/projects.rake]: the comment makes me wonder what the callbacks might do and whether it's important

File: lib/tasks/projects.rake (hunk 1 above)
Comment (mikeyoung85) [reply lib/tasks/projects.rake]: Need to fix up `NuGet`

File: lib/tasks/projects.rake (hunk 2 above)
Comment (mikeyoung85) [reply lib/tasks/projects.rake]: The callbacks will be doing a lot, but nothing related to this value so I don't think it is worth running them. On save the Project will update it's Repository which also has callbacks, update the source rank, set up dependencies for the project, call all these methods on Project:
```
normalize_licenses
set_latest_release_published_at
set_latest_release_number
set_latest_stable_release_info
set_runtime_dependencies_count
set_language
```

**Pull request merged**